package com.neg.technology.human.resource.employee.model.request;

import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class CreateEmployeeRequest {
    @NotNull(message = "Person ID is required")
    private Long personId;

    @NotNull(message = "Department ID is required")
    private Long departmentId;

    @NotNull(message = "Position ID is required")
    private Long positionId;

    @NotNull(message = "Company ID is required")
    private Long companyId;

    private Long managerId;

    private String registrationNumber;

    @NotNull(message = "Hire date is required")
    private LocalDateTime hireDate;

    @NotNull(message = "Employment start date is required")
    private LocalDateTime employmentStartDate;

    private LocalDateTime employmentEndDate;

    @NotNull(message = "isActive flag is required")
    private Boolean isActive;
}
